From 7930792e0bd86a1c3d6469f63a3fa0cc39cc0f8f Mon Sep 17 00:00:00 2001
From: stickz <stickman002@mail.com>
Date: Sat, 30 Dec 2023 06:59:53 -0500
Subject: [PATCH] Fix dynamic commands memory leak

Resolves a memory leak with dynamic commands in the .rtorrent.rc file.
---
 src/command_dynamic.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/command_dynamic.cc b/src/command_dynamic.cc
index a8d0ff02f..3d7a123b1 100644
--- a/src/command_dynamic.cc
+++ b/src/command_dynamic.cc
@@ -147,7 +147,7 @@ system_method_insert_object(const torrent::Object::list_type& args, int flags) {
     throw torrent::input_error("Invalid type.");
   }
 
-  int cmd_flags = 0;
+  int cmd_flags = rpc::CommandMap::flag_delete_key;
 
   if (!(flags & rpc::object_storage::flag_static))
     cmd_flags |= rpc::CommandMap::flag_modifiable;
